NotadogApi/Infrastructure/CurrentUserAccessor.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Linq;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Http;
+using System.IdentityModel.Tokens.Jwt;
+
+namespace NotadogApi.Infrastructure
+{
+    public class CurrentUserAccessor : ICurrentUserAccessor
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public CurrentUserAccessor(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        public int GetCurrentId()
+        {
+            return Int32.Parse(_httpContextAccessor.HttpContext.User?.Claims?.FirstOrDefault(x => x.Type == ClaimTypes.NameIdentifier)?.Value);
```

Comment:
Too hard to read, cause of line length. Consider changing to multi-line statement or using extra variable (ex. claims).